src/transformers/modeling_utils.py

```diff
@@ -2186,6 +2188,18 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                 "The argument `trust_remote_code` is to be used with Auto classes. It has no effect here and is"
                 " ignored."
             )
+
+        # change device_map into a map if we passed an int, a str or a torch.device
+        if isinstance(device_map, torch.device):
+            device_map = {"": device_map.index}
+        elif isinstance(device_map, str) and device_map not in ["auto", "balanced", "balanced_low_0", "sequential"]:
+            device_map = {"": torch.device(device_map).index}
+        elif isinstance(device_map, int):
+            if device_map < 0:
+                device_map = {"": "cpu"}
```

Comment:
I would raise a ValueError here. PyTorch does not accept negative ints in `torch.device` so we shouldn't either.